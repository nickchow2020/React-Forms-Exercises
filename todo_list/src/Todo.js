import React from "react"
import "./Todo.css"
const Todo = ({todo,removeTodo})=>{

    const deleteTodo = ()=>{
        removeTodo(todo)
    }

    return (
        <div>
            <h3>{todo}<button onClick={deleteTodo} className="btn">X</button></h3>
        </div>
    )
}

export default Todo