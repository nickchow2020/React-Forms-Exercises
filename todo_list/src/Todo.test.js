import { render,fireEvent} from '@testing-library/react';
import Todo from './Todo';

test("should it render",()=>{
    render(<Todo />)
})

test("should it match snapshot",()=>{
    const {asFragment} = render(<Todo />)
    expect(asFragment()).toMatchSnapshot()
})

test("should having content",()=>{
    const {getByText} = render(<Todo todo="gym"/>)
    expect(getByText("gym")).toBeInTheDocument()
})

test("should having delete btn",()=>{
    const {getByText} = render(<Todo todo="gym"/>)
    expect(getByText("X")).toBeInTheDocument()
})