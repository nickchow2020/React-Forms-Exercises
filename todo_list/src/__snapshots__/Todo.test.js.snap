// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should it match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      <button
        class="btn"
      >
        X
      </button>
    </h3>
  </div>
</DocumentFragment>
`;
